controllers/imagecollector/imagecollector_controller.go

```diff
@@ -0,0 +1,324 @@
+		ch <- event.GenericEvent{
+			Object: &eraserv1alpha1.ImageCollector{},
+		}
+		log.Info("Queued first ImageCollector reconcile")
+	}()
+
+	return nil
+}
+
+//+kubebuilder:rbac:groups=eraser.sh,resources=imagecollectors,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=eraser.sh,resources=imagecollectors/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=eraser.sh,resources=imagecollectors/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the ImageCollector object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.2/pkg/reconcile
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log.Info("ImageCollector Reconcile")
+
+	imageCollector := &eraserv1alpha1.ImageCollector{
+		TypeMeta:   metav1.TypeMeta{Kind: "ImageCollector", APIVersion: "eraser.sh/v1alpha1"},
+		ObjectMeta: metav1.ObjectMeta{Name: "imagecollector-shared"},
+
+		Spec: eraserv1alpha1.ImageCollectorSpec{Images: []eraserv1alpha1.Image{}},
+	}
+
+	if err := r.Get(ctx, types.NamespacedName{Name: "imagecollector-shared", Namespace: "default"}, imageCollector); err != nil {
+		if isNotFound(err) {
+			if err := r.Create(ctx, imageCollector); err != nil {
+				log.Info("could not create shared image collector")
+				return reconcile.Result{}, err
+			}
+		} else {
+			log.Info("could not get shared image collector")
+			return reconcile.Result{}, err
+		}
+	}
+
+	log.Info("xyz", "imageCollector", imageCollector)
+
+	imageJobList := &eraserv1alpha1.ImageJobList{}
+	if err := r.List(ctx, imageJobList); err != nil {
+		log.Info("could not list imagejobs")
+		return reconcile.Result{}, err
+	}
+
+	relevantJobs := util.FilterJobListByOwner(
+		imageJobList.Items, metav1.NewControllerRef(imageCollector, schema.GroupVersionKind{
+			Group:   "eraser.sh",
+			Version: "v1alpha1",
+			Kind:    "ImageCollector",
+		}),
+	)
+
+	if len(relevantJobs) > 1 {
+		return reconcile.Result{}, fmt.Errorf("More than one collector ImageJob scheduled")
+	}
+	if len(relevantJobs) == 0 {
+		if res, err := r.createImageJob(ctx, req, imageCollector); err != nil {
+			return res, err
+		}
+		return ctrl.Result{RequeueAfter: time.Second * 30}, nil
+	}
+
+	// else length is 1, so check job phase
+	switch phase := relevantJobs[0].Status.Phase; phase {
+	case eraserv1alpha1.PhaseCompleted:
+		log.Info("completed phase")
+		if res, err := r.updateSharedCRD(ctx, req, imageCollector); err != nil {
+			return res, err
+		}
+		if _, err := r.handleJobDeletion(ctx, &relevantJobs[0]); err != nil {
```

Comment:
This has problems. We're discarding the `ctrl.Result`, but the `ctrl.Result` has useful information in it (see line 219). There are two kinds of requeue we're looking at:
1. Requeue if there's a deletion delay
2. Requeue on a timer

we need to be able to handle both here.